src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -52,31 +52,49 @@ private Editor_InProc()
         public static Editor_InProc Create()
             => new Editor_InProc();
 
+        protected override bool HasActiveTextView()
+            => ErrorHandler.Succeeded(TryGetActiveTextViewHost().hr);
+
         protected override IWpfTextView GetActiveTextView()
-            => GetActiveTextViewHost().TextView;
+        {
+            return GetActiveTextViewHost().TextView;
+        }
 
         private static IVsTextView GetActiveVsTextView()
         {
-            var vsTextManager = GetGlobalService<SVsTextManager, IVsTextManager>();
+            var (textView, hr) = TryGetActiveVsTextView();
+            Marshal.ThrowExceptionForHR(hr);
+            return textView;
+        }
 
+        private static (IVsTextView textView, int hr) TryGetActiveVsTextView()
+        {
+            var vsTextManager = GetGlobalService<SVsTextManager, IVsTextManager>();
             var hresult = vsTextManager.GetActiveView(fMustHaveFocus: 1, pBuffer: null, ppView: out var vsTextView);
-            Marshal.ThrowExceptionForHR(hresult);
-
-            return vsTextView;
+            return (vsTextView, hresult);
         }
 
         private static IWpfTextViewHost GetActiveTextViewHost()
+        {
+            var (textViewHost, hr) = TryGetActiveTextViewHost();
+            Marshal.ThrowExceptionForHR(hr);
+            return textViewHost;
+        }
+
+        private static (IWpfTextViewHost textViewHost, int hr) TryGetActiveTextViewHost()
         {
             // The active text view might not have finished composing yet, waiting for the application to 'idle'
             // means that it is done pumping messages (including WM_PAINT) and the window should return the correct text view
             WaitForApplicationIdle(Helper.HangMitigatingTimeout);
 
-            var activeVsTextView = (IVsUserData)GetActiveVsTextView();
-
-            var hresult = activeVsTextView.GetData(IWpfTextViewId, out var wpfTextViewHost);
-            Marshal.ThrowExceptionForHR(hresult);
+            var (activeVsTextView, hr) = TryGetActiveVsTextView();
```

Comment:
ah, is it not to throw when view is not there? and just return false so that it can retry?

Reply:
hmm.. no you still throw.

Reply:
I'll take a closer look tomorrow and make sure this is doing what I intended before merge.

Reply:
I verified that the `Try*` methods no longer throw, and that `HasActiveTextView` does not throw. Let me know if you have other questions and I'm happy to explain further (some of this relies on the behavior of VS APIs outside of normal Roslyn code).